File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/common/SwipeableSpan.java
package ru.mamykin.foreignbooksreader.ui.common;

import android.text.style.ClickableSpan;
import android.view.View;

/**
 * Класс для расширения Span лонг тапом и горизонтальными свайпами
 */
public abstract class SwipeableSpan extends ClickableSpan {
    abstract public void onLongClick(View view);

    abstract public void onSwipeLeft(View view);

    abstract public void onSwipeRight(View view);
}

File: app/src/main/java/ru/mamykin/foboreader/ui/global/widget/SwipeableSpan.kt
package ru.mamykin.foboreader.ui.global.widget

import android.text.style.ClickableSpan
import android.view.View

/**
 * Класс для расширения Span лонг тапом и горизонтальными свайпами
 */
abstract class SwipeableSpan : ClickableSpan() {
    abstract fun onLongClick(view: View)

    abstract fun onSwipeLeft(view: View)

    abstract fun onSwipeRight(view: View)
}